src/components/ProposalMeta.tsx

```diff
@@ -12,7 +13,19 @@ const FileTypeIconMap: Record<string, IconDefinition> = {
   'application/pdf': faFilePdf,
 }
 export default function ProposalMeta({ proposalDetails }: ProposalMetaProps) {
+  const { data: session } = useSession()
+
   if (!proposalDetails) return null
+
+  const getSupervisedBy = () => {
```

Comment:
perfect use case for https://react.dev/reference/react/useMemo -> the computation is not expensive per se, but it could become more expensive, and good to understand why use memo could be valuable in such cases

Reply:
@mxmlnwbr yes, it requires a reordering to conform with the hooks rule -> pushed a commit that shows an example that should work. PR should be ready for merge :)